File: 2020/01/01-2.rb
input = File.open('input.txt').map(&:to_i)

for i in 0..input.size-1 do
  for j in i+1..input.size-1 do
    for k in j+1..input.size-1 do
      if input[i] + input[j] + input[k] == 2020
        puts input[i] * input[j] * input[k]
        return
      end
    end
  end
end

File: 2015/01/01-2.py
# Python 3.10

floor = 0

with open('input.txt') as input:
    for position, char in enumerate(input.read()):
        match char:
            case '(': floor += 1
            case ')': floor -= 1

        if floor < 0:
            print(position + 1)
            break
